x/distribution/keeper/grpc_query.go

```diff
@@ -0,0 +1,178 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+	"github.com/cosmos/cosmos-sdk/x/staking/exported"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var _ types.QueryServer = Keeper{}
+
+// Params queries params of distribution module
+func (k Keeper) Params(c context.Context, req *types.QueryParamsRequest) (*types.QueryParamsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	var params types.Params
+	k.paramSpace.GetParamSet(ctx, &params)
+
+	return &types.QueryParamsResponse{Params: params}, nil
+}
+
+// ValidatorOutstandingRewards queries rewards of a validator address
+func (k Keeper) ValidatorOutstandingRewards(c context.Context, req *types.QueryValidatorOutstandingRewardsRequest) (*types.QueryValidatorOutstandingRewardsResponse, error) {
+	if req.String() == "" || req.ValidatorAddress.Empty() {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid request")
+	}
```

Comment:
Can we update all the queries to return a more specific error message?
```suggestion
	if req == nil {
		return nil, status.Error(codes.InvalidArgument, "empty request")
	}

	if req.ValidatorAddress.Empty() {
		return nil, status.Error(codes.InvalidArgument, "empty validator address")
	}
```